ui/lib/kmip/addon/routes/role.js

```diff
@@ -1,26 +1,26 @@
 import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
 
-export default Route.extend({
-  store: service(),
-  secretMountPath: service(),
-  pathHelp: service(),
+export default class KmipRoleRoute extends Route {
+  @service store;
+  @service secretMountPath;
+  @service pathHelp;
 
-  beforeModel() {
-    return this.pathHelp.getNewModel('kmip/role', this.secretMountPath.currentPath);
-  },
+  async beforeModel() {
```

Comment:
We don't need to add the async and await, just return the promise and it should work fine 👍

Reply:
Ah, that's right. The model hook should handle it.